packages/map/src/polygon/index.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Layer, Source } from 'react-mapbox-gl';
+import { getFillPaint, getLinePaint, getTypeProp, getGeoJSON } from './utils';
+import { useChildrenBounds } from '../map/hooks';
+import { getID } from '../utils';
+
+
+const Polygon = (props) => {
+  const {
+    type,
+    area,
+    onClick,
+  } = props;
+
+  useChildrenBounds(area);
+
+  const sourceId = useRef(getID()).current;
+  const layerId = useRef(getID()).current;
```

Comment:
tbh still looks like a package level hook would make sense? instead of import `getID` could import `useUniqueId()`